openupgrade_scripts/scripts/hr/16.0.1.1/post-migration.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Coop IT Easy (https://coopiteasy.be)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from openupgradelib import openupgrade
+
+_logger = logging.getLogger(__name__)
+
+
+def warn_about_dataloss(cr, source_relation_table, relation_comodel_field):
+    """Warn user about data loss when migrating data from many2many to
+    many2one.
+
+    :param source_relation_table: The many2many relation table
+    of the model that will on the 'one' side of the relation
+    :param relation_comodel_field: The name of the column containing ids
+    of the 'many' part of the new relation.
+    """
+    openupgrade.logged_query(
+        cr,
+        """
+        SELECT DISTINCT %(relation_comodel_field)s
+        FROM %(source_relation_table)s
+        WHERE %(relation_comodel_field)s IN (
+            SELECT %(relation_comodel_field)s
+            FROM %(source_relation_table)s
+            GROUP BY %(relation_comodel_field)s
+            HAVING COUNT(*) > 1
+        )
+        """
+        % {
+            "source_relation_table": source_relation_table,
+            "relation_comodel_field": relation_comodel_field,
+        },
+    )
```

Comment:
Using the escaping method to give parameters to the query, leads to a error in the syntax of the query.

```python
openupgrade.logged_querry(
    cr,
    "SELECT * FROM %(source_relation_table)s;",
    {"source_relation_table": "hr_plan_hr_plan_activity_type_rel"},
)
```
Give an error with the following query:
```sql
SELECT * FROM 'hr_plan_hr_plan_activity_type';
```
```
ERROR:  syntax error at or near "'hr_plan_hr_plan_activity_type_rel'"
LINE 1: SELECT * FROM 'hr_plan_hr_plan_activity_type_rel';
```

So I formatted the query string with string format mechanism, which is in general a bad idea.

Reply:
I think using `AsIs` from `psycopg2.extensions` can help you. If I am not mistaking:

from pyscopg2.extensions import AsIs

openupgrade.logged_query( cr, "SELECT * FROM %(source_relation_table)s;", {"source_relation_table": AsIs("hr_plan_hr_plan_activity_type_rel")}, )

Reply:
@marielejeune Thank you for the advice about `AsIs`. It works well. :)